app/src/main/java/com/thanksmister/iot/mqtt/alarmpanel/ui/views/BaseAlarmView.kt

```diff
@@ -101,9 +106,53 @@ abstract class BaseAlarmView : LinearLayout {
         }
     }
 
+    override fun onVisibilityChanged(changedView: View?, visibility: Int) {
+        super.onVisibilityChanged(changedView, visibility)
+
+        if (!this.isShown){
+            stopFingerprintIdentity()
+            return
+        }
+
+        startFingerprintIdentity()
+    }
+
+
     override fun onDetachedFromWindow() {
         super.onDetachedFromWindow()
         destroySoundUtils()
+        stopFingerprintIdentity()
+    }
+
+    private fun startFingerprintIdentity(){
+        Timber.d("Fingerprint identity start");
+
+        fingerPrintIdentity.startIdentify(BaseAlarmView.Companion.MAX_FINGERPRINT_RETRIES, object : BaseFingerprint.FingerprintIdentifyListener{
+            override fun onSucceed(){
+                Timber.d("Fingerprint identity success");
+
+                codeComplete = false;
+                enteredCode = ""
+                addPinCode(currentCode.toString().padStart(BaseAlarmView.Companion.MAX_CODE_LENGTH, '0'))
+            }
+
+            override fun onNotMatch(availableTimes: Int) {
+                Timber.d("Fingerprint identity no match");
```

Comment:
Perhaps here we could message user (w/toast) that the fingerprint didn't match.  Not a requirement for this feature, just a suggestion.